### ISSUE TITLE
Why does mish return significantly worse result than elu?

### ISSUE THREAD

**Issue by qo4on** [digantamisra98/Mish#28]
I've tested mish vs elu on a simple feedforward network with L2 and dropout and got this result:
![download](https://user-images.githubusercontent.com/35609308/73350259-82894b80-42ae-11ea-9137-fb2eb0445089.png)

You can check it with [Colab Notebook](https://colab.research.google.com/drive/1NS1oNyvXf3M-m7t6WWWXFYviA6dia7P6), press ctrl+f9.

**Comment by digantamisra98** [comment]
Hey @qo4on. Thanks for providing me the notebook, would you mind running Mish for the complete number of Epochs as ELU was run. Additionally, if you could, run it for 3 runs to get the P-statistic to see how statistically different they're. So far, in my testing, especially in the domain of vision, Mish has consistently proved superior to ELU. Clearly, I have to do many more tests. Thanks for the heads up.

**Comment by qo4on** [comment]
I ran this test several times and didn't see much difference compared to the picture above. Longer training leads to overfitting:
![download](https://user-images.githubusercontent.com/35609308/73433229-31d52980-4366-11ea-968c-5c21286cd717.png)
Do you have any idea why this is happening?

**Comment by digantamisra98** [comment]
@qo4on I have to investigate this much further, don't have a solid conclusion on this yet. Did you try Swish or ReLU?

**Comment by qo4on** [comment]
I tried ReLU. It works even worse, overfits quickly.
![download](https://user-images.githubusercontent.com/35609308/73441927-aa8fb200-4375-11ea-9172-7ea0eeb3ee6c.png)
If you find out what's going on please don't forget to post here. It's very interesting behaviour.

**Comment by digantamisra98** [comment]
That's weird. Haven't seen such an issue in any of my experiments so far. I'll investigate further.

**Comment by LifeIsStrange** [comment]
@digantamisra98 why was it closed? Did you found an explanation?

**Comment by digantamisra98** [comment]
@LifeIsStrange I am working on the theoretical reasoning behind this and should be able to get something substantial out by the next version release but until then this issue was closed because it was stale.

**Comment by LifeIsStrange** [comment]
Any expected timeframe for the next version release? (take your time :})

**Comment by digantamisra98** [comment]
Maybe next month since I'm busy with some other commitments.
